paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -106,16 +105,10 @@ public InnerStreamTableScan newStreamScan() {
 
     protected abstract FileStoreTable copy(TableSchema newTableSchema);
 
+    // sometimes we have to change some Immutable options to implement features, so don't check it
     @Override
     public FileStoreTable copy(Map<String, String> dynamicOptions) {
-        // check option is not immutable
         Map<String, String> options = new HashMap<>(tableSchema.options());
-        dynamicOptions.forEach(
```

Comment:
don't remove this, create an another internal method.

Reply:
You mean in this internal method, we can still check immutable options changing except merge engine?

Reply:
I mean create a new method `internalCopyWithoutCheck`.